### DISCUSSION TITLE
Get to understand ECO mode - especially discharge

### DISCUSSION THREAD

**Discussion by feutl** [mletenay/home-assistant-goodwe-inverter#242] [Q&A]
I try to understand Eco mode and therefor I need some help. I read some threads on some forums on the internet but opinions do not match within the forums or with my investigation.

Lets start with the first assumption I had based on reading "the internet" ;)

(1)
For now we assume that there is no solar production happening or available. I switch ECO mode on and define the times to charge and discharge my battery. In my particular case in winter charging should start at 23:00 to 7:00 and from there I define the discharge slot.
Both slots have a percentage to be set, which defines the percentage of the maximum charge/discharge capabilities of the battery.

Now here is the second I read in a forum but did not work as described, so either I did something wrong, or "the internet" was wrong
(2)
If I set the export limit to "0" the discharge of the battery will match the house consumption based on the smart meter connected to the inverter (like in general mode).

As I said, (2) did not work for me, I switched to general mode and the discharge power of the battery matched the house consumption.

I did all this changes via SEMS App and the its possibilities as for this scenario it offered the same option as PV Master. Anyhow, should this (ECO discharge matching house consumption) work, if yes what did I do wrong :)

Thanks

**Comment by lampra1** [comment]
> Both slots have a percentage to be set, which defines the percentage of the maximum charge/discharge capabilities of the battery.

This is not the case, the percentage refers to the inverter AC rating. For example, 30% for a 5Kw inverter would be 1500w.

> If I set the export limit to "0"

 Again, this is not the case, you need to set eco charge or discharge percentage at 0% to emulate general mode. 
Remember that at this setting, sometimes the export/import might not be exact for small periods. For 3 phase inverters, it also depends on the phase and the export limit but for trivial amounts of energy.
